features/admin.verifiable-credentials.v1/components/vc-template-list.tsx

```diff
@@ -0,0 +1,278 @@
+    TableActionsInterface,
+    TableColumnInterface
+} from "@wso2is/react-components";
+import React, { ReactElement, ReactNode, SyntheticEvent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Header, Icon, SemanticICONS } from "semantic-ui-react";
+import { deleteVCTemplate } from "../api/verifiable-credentials";
+import { VCTemplateListItem } from "../models/verifiable-credentials";
+
+interface VCTemplateListProps extends IdentifiableComponentInterface {
+    advancedSearch: ReactNode;
+    isLoading: boolean;
+    mutateTemplateList: () => void;
+    list: VCTemplateListItem[];
+    onSearchQueryClear?: () => void;
+    searchQuery?: string;
+    setShowAddTemplateWizard: () => void;
+}
+
+/**
+ * Verifiable Credentials template list component.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const VCTemplateList = ({
+    advancedSearch,
+    isLoading,
+    list,
+    mutateTemplateList,
+    onSearchQueryClear: _onSearchQueryClear,
+    searchQuery: _searchQuery,
+    setShowAddTemplateWizard,
+    ["data-componentid"]: componentId
+}: VCTemplateListProps): ReactElement => {
+    const dispatch: any = useDispatch();
+    const { t } = useTranslation();
+
+    const [ showDeleteConfirmation, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ currentDeletedTemplate, setCurrentDeletedTemplate ] = useState<VCTemplateListItem>();
+
+    /**
+     * Handles template deletion.
+     *
+     * @param template - Template to be deleted.
+     */
+    const handleTemplateDelete = (template: VCTemplateListItem): void => {
+        deleteVCTemplate(template.id)
+            .then(() => {
+                dispatch(
+                    addAlert({
+                        description: t(
+                            "verifiableCredentials:notifications.deleteTemplate.success.description"
+                        ),
+                        level: AlertLevels.SUCCESS,
+                        message: t("verifiableCredentials:notifications.deleteTemplate.success.message")
+                    })
+                );
+                mutateTemplateList();
+            })
+            .catch(() => {
+                dispatch(
+                    addAlert({
+                        description: t(
+                            "verifiableCredentials:notifications.deleteTemplate.error.description"
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t("verifiableCredentials:notifications.deleteTemplate.error.message")
+                    })
+                );
+            });
+    };
+
+    /**
+     * Resolves table actions.
+     *
+     * @returns Table actions.
+     */
+    const resolveTableActions = (): TableActionsInterface[] => {
+        return [
+            {
+                "data-testid": `${componentId}-item-edit-button`,
```

Comment:
shall we check whether we could use data-componentid. We no longer use data-testid.